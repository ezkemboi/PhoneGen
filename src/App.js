import React from 'react';
import './App.scss';

class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      numbersGenerated: [1, 2, 3],
      generatedPhoneNumber: '',
      amount: 0
    }
  }

  componentDidMount() {
    // Fetch stored phone numbers from local storage

  }

  // handleChange = (e) => {
  //   e.preventDefault();
  //   this.setState({
  //     amount: amount
  //   })
  // }

  // Generate new phone number function
  generateNewPhoneNumber = () => {
    // Generate 9 random numbers
    const { numbersGenerated, amount } = this.state;
    // for(i = 0; i <= amount; i++) {
    //   const number = `0${Math.floor(Math.random() * 1000000000)}`;
    // }
    let number;
    number = `0${Math.floor(Math.random() * 1000000000)}`;
    // Check whether the number already exist
    if (!numbersGenerated.includes(number)) {
      numbersGenerated.push(number);
      // Set the current generated numbers(Add 0 at the beginning)
      this.setState({
        generatedPhoneNumber: number
      });
    }
  }

  render() {
    const { generatedPhoneNumber, amount } = this.state;
    return (
      <div className="App">
        <h3>New Generator APP</h3>
        <p>
          Welcome to new phone number generator app.
          Genrate and store your phone numbers for retrival
        </p>
        <div>
          <form >
            <input name="phonenumber" name={amount} />
            <button id="generate-numbers" onClick={() => this.generateNewPhoneNumber()}>Generate</button>
          </form>
        </div>
        <p>Current Phone Number: <b>{generatedPhoneNumber}</b></p>
        <h3>Check New Phone Number</h3>
        <div className="Download-pdf-file">
          <button>Download As PDF</button>
        </div>
        <div className="Phone-Number-table">
          <table>
            <thead>
              <tr>
                <th>Phone Numbers</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>07893637838</td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    )
  }
}

export default App;
