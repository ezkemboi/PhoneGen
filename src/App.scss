.App {
  display: flex;
  margin: 2% 10%;
  min-height: 85vh;
  max-height: 90vh;
  text-align: center;
  &__left {
    display: flex;
    flex-direction: column;
    width: 30%;
    margin-right: 30px;
    background-color: white;
    border: 1px solid gray;
    h3 {
      border-bottom: 1px solid currentColor;
      padding-bottom: 20px;
      width: 50%;
      align-self: center;
    }
    #app-description {
      margin: 20px;
      padding: 0 10px;
    }
    input {
      height: 30px;
      margin: 10px 20px;
      width: 40%;
      align-self: center;
      text-align: center;
    }
    button {
      display: flex;
      align-self: center;
      width: 70%;
      justify-content: center;
    }
    button:disabled:hover {
      background-color: white;
      color: black;
      opacity: 0.3;
    }
  }
  &__right {
    display: flex;
    background-color: white;
    width: 70%;
    flex-direction: column;
    border: 1px solid gray;
    .right-header-section {
      display: flex;
      justify-content: flex-end;
      margin: 10px;
      .clear-button {
        margin-right: 10px;
        border: 2px solid black;
        border-radius: 8px;
      }
      .clear-button:hover {
        background-color: red;
      }
      .order-contacts {
        display: flex;
        margin-right: 10px;
        align-self: center;
        font-size: 16px;
        border: 2px solid black;
        border-radius: 8px;
        height: 40px;
      }
    }
    .headers-for-phones {
      display: flex;
      border-bottom: 1px solid gray;
      border-top: 1px solid gray;
      &__index {
        width: 30%;
      }
      &__number {
        width: 70%;
      }
    }
    .contacts-table-list {
      overflow-y: scroll;
      .phone-list-table {
        display: flex;
        padding: 0 20px;
        &__left {
          width: 30%;
          p {
            border-bottom: 1px solid gray;
            padding-bottom: 2px;
          }
        }
        &__right {
          width: 70%;
          p {
            border-bottom: 1px solid gray;
            padding-bottom: 2px;
          }
        }
      }
    }
  }
  button {
    background-color: white;
    height: 40px;
    color: black;
    font-size: 16px;
  }
  button:hover {
    cursor: pointer;
    background: black;
    color: white;
  }
}

#app-title {
  text-align: center;
  font-size: 30px;
  line-height: 20px;
}
