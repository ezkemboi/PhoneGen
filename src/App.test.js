import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('APP', () => {
  it('renders without crashing', () => {
    const component = shallow(<App />)
    component.find('button#generate-numbers').simulate('click');
    expect(component).toMatchSnapshot();
    expect(component.length).toEqual(1)
  })
})
