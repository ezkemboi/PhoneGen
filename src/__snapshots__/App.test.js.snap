// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP renders without crashing 1`] = `ShallowWrapper {}`;
